command/agent/consul/service_client_test.go

```diff
@@ -578,6 +578,7 @@ func TestSyncLogic_proxyUpstreamsDifferent(t *testing.T) {
 		return api.Upstream{
 			Datacenter:       "sfo",
 			DestinationName:  "billing",
+			DestinationType:  "service",
```

Comment:
I think we should still have a test where the `DestinationType` is `""` on the Nomad side and non empty on the Consul side